### PR TITLE
Test spectroscopic axis

### PR DESCRIPTION
**Pull request by dinossimpson**
Added test_spectroscopicAxis.py and test_Cube_fit.py
test_spectroscopicAxis.py runs a loop creating spectroscopicAxis instances with fake data+noise,
runs Spectrum.specfit on them and asserts that the resulting parameters are close to the Axis' parameters.
test_Cube_fit.py follows the same concept but instantiates SpectralCubes instead and runs their fiteach() method.

### CODE REVIEW

File: test_Cube_fiteach.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+						passes['avg_difference'][k] += abs(result_cube[:,i,j][k] - params[k]).tolist()
+				else:
+					for k in range(3):
+						fails['fails'][k]+=1
+						fails['avg_difference'][k] += abs(result_cube[:,i,j][k] - params[k]).tolist()
+				assertion_list.append(assertion)
+
+	print 'assertion_list:', np.array(assertion_list)
+	print 'passes:', passes
+	print 'fails:', fails
```
Comment (keflavich) [inline test_Cube_fiteach.py]: since this is a test, we do want the final action to be a `raise` if there are failures

File: test_moments.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+for model_name in sp.specfit.Registry.multifitters.iterkeys():
+    try:
+        print 'testing:', model_name
+        model = sp.specfit.Registry.multifitters[model_name]
+        moments = getattr(model, 'moments')
+        rawdata = np.random.randn(xarr.size)
+        params = model.moments(xarr, rawdata)
+        print 'params from moments:',params
+        xarr = SpectroscopicAxis(xarr)
+        model.n_modelfunc(pars=params)(xarr)
```
Comment (keflavich) [inline test_moments.py]: (remember, comment here)

File: test_Cube_fiteach.py (hunk 1 above)
Comment (dinossimpson) [inline test_Cube_fiteach.py]: I've added an assert result < 2 on line 33, should I raise and Exception instead with a message?

**Comment by dinossimpson** [comment]
I pushed again removing the test_moments commit since it is included in the other PR.
I should mention that these 2 tests are handling SpectroscopicAxis as a Quantity so they will fail on master.

File: test_Cube_fiteach.py (hunk 1 above)
Comment (keflavich) [inline test_Cube_fiteach.py]: No, for tests a simple assertion is fine, as long as it's clear what the test is doing.  Clear error messages are meant for users of the code; tests are really more to make sure it works and we understand why it doesn't if it doesn't

**Comment by keflavich** [comment]
Now that units->Quantity is in master, these should pass, right?

**Comment by keflavich** [comment]
Looks like there's a merge conflict after I merged #2 and #6 - can you fix this?

**Pull request closed without merging**